lib/chess_club/user_manager.ex

```diff
@@ -0,0 +1,124 @@
+defmodule ChessClub.UserManager do
+  @moduledoc """
+  The UserManager context.
+  """
+
+  import Ecto.Query, warn: false
+  alias ChessClub.Repo
+
+  alias ChessClub.UserManager.User
+
+  alias Argon2
+  import Ecto.Query, only: [from: 2]
+
+  def authenticate_user(username, plain_text_password) do
+    query = from u in User, where: u.username == ^username
+
+    case Repo.one(query) do
+      nil ->
+        Argon2.no_user_verify()
+        {:error, :invalid_credentials}
+
+      user ->
+        if Argon2.verify_pass(plain_text_password, user.password) do
+          {:ok, user}
+        else
+          {:error, :invalid_credentials}
+        end
+    end
+  end
+
+  @doc """
+  Returns the list of users.
+
+  ## Examples
+
+      iex> list_users()
+      [%User{}, ...]
+
```

Comment:
We should use `doctest` here, it will ensure that the examples in documentation are correct perpetually.

Reply:
Good call out. I'll look into doing this.

Reply:
I looked into this and was getting a weird cyclical error. I have extracted this card to represent the work: https://trello.com/c/TIpQNmlS/22-use-doctest-for-usermanager-module